x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+      await retry.tryForTime(30 * 1000, async () => {
+        const panel = await find.existsByCssSelector('euiContextMenuPanel');
+        if (!panel) {
+          await testSubjects.click(this.rowSelector(fieldName, 'euiCollapsedItemActionsButton'));
+          await find.existsByCssSelector('euiContextMenuPanel');
+        }
+      });
+    }
+
+    public async ensureActionsMenuClosed(fieldName: string, action: string) {
+      await retry.tryForTime(30 * 1000, async () => {
+        if (await testSubjects.exists(action)) {
+          await testSubjects.click(this.rowSelector(fieldName, 'euiCollapsedItemActionsButton'));
+          await testSubjects.missingOrFail(action, { timeout: 5000 });
+        }
+      });
+    }
+
+    public async assertActionMenuViewInLensEnabled(fieldName: string, expectedValue: boolean) {
+      await retry.tryForTime(30 * 1000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+        const actionMenuViewInLensButton = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionViewInLensButton"][class="euiContextMenuItem"]'
+        );
+        const isEnabled = await actionMenuViewInLensButton.isEnabled();
+        expect(isEnabled).to.eql(
+          expectedValue,
+          `Expected "Explore in lens" action menu button for '${fieldName}' to be '${
+            expectedValue ? 'enabled' : 'disabled'
+          }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+        );
+      });
+    }
+
+    public async assertActionMenuDeleteIndexPatternFieldButtonEnabled(
+      fieldName: string,
+      expectedValue: boolean
+    ) {
+      await this.ensureActionsMenuOpen(fieldName);
+      const actionMenuViewInLensButton = await find.byCssSelector(
+        '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+      );
+      const isEnabled = await actionMenuViewInLensButton.isEnabled();
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Delete index pattern field" action menu button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
+    }
+    public async clickActionMenuDeleteIndexPatternFieldButton(fieldName: string) {
+      const testSubj = 'dataVisualizerActionDeleteIndexPatternFieldButton';
+      await retry.tryForTime(5000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+
+        const button = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+        );
+        await button.click();
+        await this.ensureActionsMenuClosed(fieldName, testSubj);
+      });
+    }
+
+    public async assertActionMenuExists(fieldName: string) {
+      const actionButton = this.rowSelector(fieldName, 'euiCollapsedItemActionsButton');
+      await testSubjects.existOrFail(actionButton);
+    }
```

Comment:
I find the method name and the actual check a bit confusing: the method name makes me think it's looking for the (opened) action menu, but actually, it asserts that the action menu button exists.
Also, it seems this method is never used, so maybe remove it completely?

Reply:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e